bentoml/marshal/dispatcher.py

```diff
@@ -55,17 +54,19 @@ def __init__(self, interval, max_size, task_concurrency=2):
         """
         self.interval = interval
         self.max_size = max_size
-        self.task_concurrency = task_concurrency
+        self.shared_sema = shared_sema
         self.callback = None
         self._current_parade = None
+        self._sema = None
 
     @property
-    @lru_cache(maxsize=1)
     def outbound_sema(self):
         '''
-        semaphore should be created after forked(in the current event loop)
+        semaphore should be created after process forked
         '''
-        return asyncio.Semaphore(self.task_concurrency)
+        if self._sema is None:
+            self._sema = self.shared_sema and self.shared_sema() or asyncio.Semaphore(1)
```

Comment:
consider this for readability?
```
self._sema = self.shared_sema() if self.shared_sema else asyncio.Semaphore(1)
```

Reply:
The later is always better than the former. I always forget it.